src/server_name.rs

```diff
@@ -257,6 +253,277 @@ enum State {
     Hyphen { len: usize },
 }
 
+/// `no_std` implementation of `std::net::IpAddr`.
```

Comment:
Once `std::net::IpAddr` and friends are moved to `core::net::IpAddr` I am hoping this can be lost.

Reply:
Yup, that's the idea.

Reply:
Perhaps the semver bump could be avoided by making the `IpAddr` value of `ServerName` private? Are there any cases where the application needs to read out that value?

Alternately: semver is about what interfaces are documented to be stable, so we can document our way out of this by saying that the type of the `IpAddr` value is explicitly not stable (yet).

Reply:
> Perhaps the semver bump could be avoided by making the `IpAddr` value of `ServerName` private? Are there any cases where the application needs to read out that value?

Well, webpki needs to match `ServerName` to get out the DNS name bytes or the IP address bytes, so I don't think hiding this type is straightforward.

> Alternately: semver is about what interfaces are documented to be stable, so we can document our way out of this by saying that the type of the `IpAddr` value is explicitly not stable (yet).

That feels like a reasonable compromise. We can say that the interfaces to build a `ServerName` are stable but that the identity of the `IpAddr` type is not.